### PR TITLE
transaction: improve executer tests

### PR DESCRIPTION
**Pull request by kopiczko**
- make table tests for testing result
- remove irrelevant assertions

### CODE REVIEW

File: transaction/executer_test.go [hunk 1]
```diff
@@ -3,7 +3,6 @@ package transaction
 import (
```
Comment (kopiczko) [inline transaction/executer_test.go]: All removed lines are replaced with this: https://github.com/giantswarm/microkit/blob/983e16add51e2027056de0cc62079dc148de35f2/transaction/executer_test.go#L235-L321

Diff is ugly.

File: transaction/executer_test.go [hunk 2]
```diff
@@ -233,363 +232,90 @@ func Test_Executer_TransactionIDGiven_NoReplay(t *testing.T) {
...
+			executeConfig.TrialID = "test-trial-ID"
 		}
-	}
 
-	{
-		err := newExecuter.Execute(ctx, executeConfig)
+		err = newExecuter.Execute(ctx, executeConfig)
 		if err != nil {
-			t.Fatal("expected", nil, "got", err)
+			t.Errorf("#%d: want nil, got %#v", i, err)
```
Comment (MarcelMue) [inline transaction/executer_test.go]: Minor issue, but please keep `expected` and not `want`. It's more clear what is meant in a test that way IMO

File: transaction/executer_test.go [hunk 3]
```diff
@@ -233,363 +232,90 @@ func Test_Executer_TransactionIDGiven_NoReplay(t *testing.T) {
...
+		TrialOutput     interface{}
+		WantReplayInput interface{}
+	}{
+		{ // 0
+			TrialOutput:     []byte("hello world"),
+			WantReplayInput: "hello world",
+		},
+		{ // 1
+			TrialOutput:     float64(4.3),
+			WantReplayInput: "4.3",
```
Comment (MarcelMue) [inline transaction/executer_test.go]: This is not part of this specific change, but why was `4.3` chosen here? Does it have any meaning or just random var?

Comment (kopiczko) [inline transaction/executer_test.go]: It's random.

File: transaction/executer_test.go (hunk 2 above)
Comment (kopiczko) [inline transaction/executer_test.go]: I'd like to keep that. This is like it is in `etcd` codebase.
I think I earned that :D

**Comment by kopiczko** [comment]
@MarcelMue do you approve?

File: transaction/executer_test.go (hunk 2 above)
Comment (MarcelMue) [inline transaction/executer_test.go]: We should aim to be consistent then, now we used 'expected' and 'want' mixed :(

Comment (xh3b4sd) [inline transaction/executer_test.go]: +1 on Marcel. Also, correct grammar would then be `wants`.

File: transaction/executer_test.go [hunk 4]
```diff
@@ -233,363 +232,90 @@ func Test_Executer_TransactionIDGiven_NoReplay(t *testing.T) {
...
-	}
-
-	{
-		err := newExecuter.Execute(ctx, executeConfig)
+func Test_Executer_TransactionResult(t *testing.T) {
+	tests := []struct {
+		TrialOutput     interface{}
+		WantReplayInput interface{}
+	}{
+		{ // 0
```
Comment (xh3b4sd) [inline transaction/executer_test.go]: I would like to see comments in a separate line and a bit more elaborative. This is pretty useless to me personally.

**Comment by xh3b4sd** [comment]
As a general note thanks for your boyscouting efforts here. More of this please. <3

File: transaction/executer_test.go (hunk 4 above)
Comment (kopiczko) [inline transaction/executer_test.go]: When something fails like this `t.Errorf("#%d: want %#v, got %#v", i, nil, replayInput)` fails then you'd see `#0: want nil, got "abc"`, and this `#0` points you to the right case. At least for me it's useful.

**Pull request merged**